pkg/composableschemadsl/compiler/compiler.go

```diff
@@ -68,12 +72,21 @@ func AllowUnprefixedObjectType() ObjectPrefixOption {
 	return func(cfg *config) { cfg.objectTypePrefix = new(string) }
 }
 
+func SourceFolder(sourceFolder string) Option {
```

Comment:
Add a doc comment explaining what this does. Probably good to add doc comments to the other exported functions too